pkg/core/stateroot/module.go

```diff
@@ -114,6 +116,42 @@ func (s *Module) Init(height uint32, enableRefCount bool) error {
 	return nil
 }
 
+// CleanStorage removes all MPT-related data from the storage (MPT nodes, validated stateroots)
+// except local stateroot for the specified height and GC flag. This method is aimed to clean
+// outdated MPT data before state sync process can be started.
+func (s *Module) CleanStorage() error {
+	gcKey := []byte{byte(storage.DataMPT), prefixGC}
+	gcVal, err := s.Store.Get(gcKey)
+	if err != nil {
+		return fmt.Errorf("failed to get GC flag: %w", err)
+	}
+	b := s.Store.Batch()
+	s.Store.Seek([]byte{byte(storage.DataMPT)}, func(k, _ []byte) {
```

Comment:
So we have an MPT for block 0. What's gonna happen to it?

Reply:
Before the state sync process can be started, all MPT-prefixed data will be removed from the storage in order to be able to store MPT nodes for the new state.

So that's the only place in the whole state sync process where we mess up the current blockchain state.

But we still keeping the current local state root in the storage to be able to stat onde normally if the state sync process will be interrupted. After state sync is completed, current local root will be replaced by the one of proper height.